app/assets/javascripts/share_item.js

```diff
@@ -0,0 +1,44 @@
+$(document).ready(function() {
+  $(document).on('click','.share_item', function(e) {
+    e.preventDefault();
+    var id = $(this).attr('data-id');
+    $(document.body).append('<div id="share_form"></div>');
+    $('#share_form').dialog({
+      open: function(){
+        fetchShareFormContent(id);
+      }
+    });
+    var $dialog = $(".ui-dialog");
+    $dialog.addClass("modal-content");
```

Comment:
all these logic to style dialog should be move to some function and not just copied.